pkg/ec2metadata/ec2metadata.go

```diff
@@ -122,6 +122,18 @@ type NodeMetadata struct {
 
 // New constructs an instance of the Service client
 func New(metadataURL string, tries int) *Service {
+	if strings.HasPrefix(metadataURL, "http://amazon-ec2-metadata-mock-service.default.svc.cluster.local") {
```

Comment:
add a comment to clarify it here or either remove it if tests are passing successfully without it

Reply:
Fixed by removing.